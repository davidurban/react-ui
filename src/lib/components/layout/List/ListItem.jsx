import PropTypes from 'prop-types';
import React from 'react';
import styles from './List.scss';

const ListItem = ({ children }) => {
  if (!children) {
    return null;
  }

  return (
    <li className={styles.item}>
      {children}
    </li>
  );
};

ListItem.defaultProps = {
  children: null,
};

ListItem.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default ListItem;
