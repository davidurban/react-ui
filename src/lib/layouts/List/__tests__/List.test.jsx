import React from 'react';
import { shallow } from 'enzyme';
import List from '../List';

describe('rendering', () => {
  it('renders correctly with no children', () => {
    const tree = shallow((
      <List />
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a single child', () => {
    const tree = shallow((
      <List>
        <span>content</span>
      </List>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple children', () => {
    const tree = shallow((
      <List>
        <span>content 1</span>
        <span>content 2</span>
        <span>content 3</span>
      </List>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple children and all props', () => {
    const tree = shallow((
      <List align="right" autoWidth>
        <span>content 1</span>
        <span>content 2</span>
        <span>content 3</span>
      </List>
    ));

    expect(tree).toMatchSnapshot();
  });
});
