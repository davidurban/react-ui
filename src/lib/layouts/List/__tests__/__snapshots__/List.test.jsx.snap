// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with a single child 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    align="left"
    autoWidth={false}
>
    <span>
        content
    </span>
</List>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list
          alignLeft"
>
        <span>
                content
        </span>
</ul>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          content
</span>,
        "className": "list
          alignLeft",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "content",
        },
        "ref": null,
        "rendered": "content",
        "type": "span",
      },
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list
          alignLeft"
>
          <span>
                    content
          </span>
</ul>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            content
</span>,
          "className": "list
          alignLeft",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content",
          },
          "ref": null,
          "rendered": "content",
          "type": "span",
        },
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`rendering renders correctly with multiple children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    align="left"
    autoWidth={false}
>
    <span>
        content 1
    </span>
    <span>
        content 2
    </span>
    <span>
        content 3
    </span>
</List>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list
          alignLeft"
>
        <span>
                content 1
        </span>
        <span>
                content 2
        </span>
        <span>
                content 3
        </span>
</ul>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            content 1
</span>,
          <span>
            content 2
</span>,
          <span>
            content 3
</span>,
        ],
        "className": "list
          alignLeft",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 1",
          },
          "ref": null,
          "rendered": "content 1",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 2",
          },
          "ref": null,
          "rendered": "content 2",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 3",
          },
          "ref": null,
          "rendered": "content 3",
          "type": "span",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list
          alignLeft"
>
          <span>
                    content 1
          </span>
          <span>
                    content 2
          </span>
          <span>
                    content 3
          </span>
</ul>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              content 1
</span>,
            <span>
              content 2
</span>,
            <span>
              content 3
</span>,
          ],
          "className": "list
          alignLeft",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 1",
            },
            "ref": null,
            "rendered": "content 1",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 2",
            },
            "ref": null,
            "rendered": "content 2",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 3",
            },
            "ref": null,
            "rendered": "content 3",
            "type": "span",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`rendering renders correctly with multiple children and all props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    align="right"
    autoWidth={true}
>
    <span>
        content 1
    </span>
    <span>
        content 2
    </span>
    <span>
        content 3
    </span>
</List>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list
          alignRight"
>
        <span>
                content 1
        </span>
        <span>
                content 2
        </span>
        <span>
                content 3
        </span>
</ul>,
      "className": "root
        isAutoWidth",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            content 1
</span>,
          <span>
            content 2
</span>,
          <span>
            content 3
</span>,
        ],
        "className": "list
          alignRight",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 1",
          },
          "ref": null,
          "rendered": "content 1",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 2",
          },
          "ref": null,
          "rendered": "content 2",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content 3",
          },
          "ref": null,
          "rendered": "content 3",
          "type": "span",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list
          alignRight"
>
          <span>
                    content 1
          </span>
          <span>
                    content 2
          </span>
          <span>
                    content 3
          </span>
</ul>,
        "className": "root
        isAutoWidth",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              content 1
</span>,
            <span>
              content 2
</span>,
            <span>
              content 3
</span>,
          ],
          "className": "list
          alignRight",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 1",
            },
            "ref": null,
            "rendered": "content 1",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 2",
            },
            "ref": null,
            "rendered": "content 2",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content 3",
            },
            "ref": null,
            "rendered": "content 3",
            "type": "span",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`rendering renders correctly with no children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    align="left"
    autoWidth={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
